pkg/code-generator/cmd/main.go

```diff
@@ -122,79 +124,115 @@ func Generate(opts GenerateOptions) error {
 	}())
 
 	var protoDescriptors []*descriptor.FileDescriptorProto
-	for _, projectConfig := range projectConfigs {
-		importedResources, err := importCustomResources(projectConfig.Imports)
+	for _, skp := range soloKitProjects {
+		importedResources, err := importCustomResources(skp.ApiGroup.Imports)
 		if err != nil {
 			return err
 		}
-
-		projectConfig.CustomResources = append(projectConfig.CustomResources, importedResources...)
-
-		for _, desc := range descriptors {
-			if filepath.Dir(desc.ProtoFilePath) == filepath.Dir(projectConfig.ProjectFile) {
-				projectConfig.ProjectProtos = append(projectConfig.ProjectProtos, desc.GetName())
+		for _, vc := range skp.ApiGroup.VersionConfigs {
+			vc.CustomResources = append(vc.CustomResources, importedResources...)
+			for _, desc := range descriptors {
+				if filepath.Dir(desc.ProtoFilePath) == filepath.Dir(skp.ProjectFile)+"/"+vc.Version {
```

Comment:
seems like we have this check in a few places. can we extract to a function?
also, use `filepath.Join` here. even though it's only required on windows. and if you use windows, you're wrong 😏